Source/SwiftLintFramework/Rules/MultilineFunctionChainsRule.swift

```diff
@@ -0,0 +1,221 @@
+        ],
+        triggeringExamples: [
+            """
+            let evenSquaresSum = [20, 17, 35, 4]
+                .filter { $0 % 2 == 0 }↓.map { $0 * $0 }
+                .reduce(0, +)
+            """,
+            """
+            let evenSquaresSum = a.b(1, 2, 3)
+                .c { blah in
+                    print(blah)
+                }↓.d()
+            """,
+            """
+            let evenSquaresSum = a.b(1, 2, 3)
+                .c(2, 3, 4)↓.d()
+            """,
+            """
+            let evenSquaresSum = a.b(1, 2, 3)↓.c { blah in
+                    print(blah)
+                }
+                .d()
+            """
+        ]
+    )
+
+    public func validate(file: File,
+                         kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        return violatingOffsets(file: file, kind: kind, dictionary: dictionary).map { offset in
+            return StyleViolation(ruleDescription: type(of: self).description,
+                                  severity: configuration.severity,
+                                  location: Location(file: file, characterOffset: offset))
+        }
+    }
+
+    private func violatingOffsets(file: File,
+                                  kind: SwiftExpressionKind,
+                                  dictionary: [String: SourceKitRepresentable]) -> [Int] {
+        let ranges = callRanges(file: file, kind: kind, dictionary: dictionary)
+
+        let calls = ranges.compactMap { range -> (dotLine: Int, dotOffset: Int, range: NSRange)? in
+            guard
+                let offset = callDotOffset(file: file, callRange: range),
+                let line = file.contents.bridge().lineAndCharacter(forCharacterOffset: offset)?.line else {
+                    return nil
+            }
+            return (dotLine: line, dotOffset: offset, range: range)
+        }
+
+        let uniqueLines = calls
+            .map { $0.dotLine }
+            .reduce(into: Set<Int>()) { (result: inout Set<Int>, line: Int) in
+                result.insert(line)
+            }
+
+        if uniqueLines.count == 1 { return [] }
+
+        // The first call (last here) is allowed to not have a leading newline.
+        let noLeadingNewlineViolations = calls
+            .dropLast()
+            .filter { line in
+                !callHasLeadingNewline(file: file, callRange: line.range)
+            }
+
+        return noLeadingNewlineViolations.map { $0.dotOffset }
+    }
+
+    private static let whitespaceDotRegex = regex("\\s*\\.")
+
+    private func callDotOffset(file: File, callRange: NSRange) -> Int? {
+        guard
+            let range = file.contents.bridge().byteRangeToNSRange(start: callRange.location, length: callRange.length),
+            case let regex = type(of: self).whitespaceDotRegex,
+            let match = regex.matches(in: file.contents, options: [], range: range).last?.range else {
```

Comment:
would using `regex.firstMatch` work here?

Reply:
We need to use the last match here to ensure that in cases where a function call follows a property access don't trigger. e.g. 
```swift
self.happeningNewsletterOn = self.updateCurrentUser
    .map { $0.newsletters.happening }.skipNil().skipRepeats()
```
If you'd like I can add a comment, but this is covered in the tests 👍